floris/turbine_library/turbine_previewer.py

```diff
@@ -0,0 +1,905 @@
+import re
+from math import ceil
+from pathlib import Path
+
+import attrs
+import matplotlib.pyplot as plt
+import numpy as np
+from attrs import define, field
+
+from floris.simulation import (
+    Ct,
+    power,
+    Turbine,
+)
+from floris.type_dec import NDArrayFloat
+from floris.utilities import load_yaml
+
+
+INTERNAL_LIBRARY = Path(__file__).parent
+DEFAULT_WIND_SPEEDS = np.linspace(0, 40, 81)
+
+
+def round_nearest_2_or_5(x: int | float) -> int:
+    """Rounds a number (with a 0.5 buffer) up to the nearest integer divisible by 2 or 5.
+
+    Args:
+        x (int | float): The number to be rounded.
+
+    Returns:
+        int: The rounded number.
+    """
+    base_2 = 2
+    base_5 = 5
+    return min(base_2 * ceil((x + 0.5) / base_2), base_5 * ceil((x + 0.5) / base_5))
+
+
+def round_nearest(x: int | float, base: int = 5) -> int:
+    """Rounds a number (with a 0.5 buffer) up to the nearest integer divisible by 5.
+
+    Args:
+        x (int | float): The number to be rounded.
+
+    Returns:
+        int: The rounded number.
+    """
+    return base * ceil((x + 0.5) / base)
+
+
+@define(auto_attribs=True)
+class TurbineInterface:
+    turbine: Turbine = field(validator=attrs.validators.instance_of(Turbine))
+
+    @classmethod
+    def from_internal_library(cls, file_name: str):
+        """Loads the turbine defintion from a YAML configuration file located in
+        ``floris/floris/turbine_library/``.
+
+        Parameters
+        ----------
+        file_name : str | Path
+            The file name of the turbine configuration file.
+
+        Returns
+        -------
+        TurbineInterface
+            Creates a new ``TurbineInterface`` object.
+        """
+        return cls(turbine=Turbine.from_dict(load_yaml(INTERNAL_LIBRARY / file_name)))
+
+    @classmethod
+    def from_yaml(cls, file_path: str | Path):
+        """Loads the turbine defintion from a YAML configuration file.
+
+        Parameters
+        ----------
+        file_path : str | Path
+            The full path and file name of the turbine configuration file.
+
+        Returns
+        -------
+        TurbineInterface
+            Creates a new ``TurbineInterface`` object.
+        """
+        return cls(turbine=Turbine.from_dict(load_yaml(file_path)))
+
+    @classmethod
+    def from_turbine_dict(cls, config_dict: dict):
+        """Loads the turbine defintion from a dictionary.
+
+        Args:
+            config_dict : dict
+                The ``Turbine`` configuration dictionary.
+
+        Returns:
+            (`TurbineInterface`): Returns a ``TurbineInterface`` object.
+        """
+        return cls(turbine=Turbine.from_dict(config_dict))
+
+    def power_curve(
+        self,
+        wind_speed: NDArrayFloat | None = None
```

Comment:
How about this: `wind_speeds: NDArrayFloat = DEFAULT_WIND_SPEEDS`, and then you can delete 130-131

Reply:
This makes a lot of sense, and so updated every instance of this usage.